# Introdução à Ciência de Dados - UFPB
Professor: Yuri Malheiros

## Medidas de Centralidade e Dispersão

A partir do dataset `movies.csv` calcule a média, mediana e desvio padrão para as colunas:

In [62]:
import pandas as pd
import numpy as np

In [4]:
filmes = pd.read_csv('C:\\Users\\Ruy\\Documents\\movies.csv')
filmes

Title    US Gross  Worldwide Gross  US DVD Sales  \
0                 The Land Girls    146083.0         146083.0           NaN   
1         First Love, Last Rites     10876.0          10876.0           NaN   
2     I Married a Strange Person    203134.0         203134.0           NaN   
3           Let's Talk About Sex    373615.0         373615.0           NaN   
4                           Slam   1009819.0        1087521.0           NaN   
...                          ...         ...              ...           ...   
3196  Zack and Miri Make a Porno  31452765.0       36851125.0    21240321.0   
3197                      Zodiac  33080084.0       83080084.0    20983030.0   
3198                        Zoom  11989328.0       12506188.0     6679409.0   
3199         The Legend of Zorro  45575336.0      141475336.0           NaN   
3200           The Mask of Zorro  93828745.0      233700000.0           NaN   

      Production Budget Release Date MPAA Rating  Running Time (min)  \
0             8000000.0    12-Jun-98           R                 NaN   
1              300000.0     7-Aug-98           R                 NaN   
2              250000.0    28-Aug-98         NaN                 NaN   
3              300000.0    11-Sep-98         NaN                 NaN   
4             1000000.0     9-Oct-98           R                 NaN   
...                 ...          ...         ...                 ...   
3196         24000000.0    31-Oct-08           R               101.0   
3197         85000000.0     2-Mar-07           R               157.0   
3198         35000000.0    11-Aug-06          PG                 NaN   
3199         80000000.0    28-Oct-05          PG               129.0   
3200         65000000.0    17-Jul-98       PG-13               136.0   

             Distributor                        Source        Major Genre  \
0               Gramercy                           NaN                NaN   
1                 Strand                           NaN              Drama   
2              Lionsgate                           NaN             Comedy   
3              Fine Line                           NaN             Comedy   
4                Trimark           Original Screenplay              Drama   
...                  ...                           ...                ...   
3196       Weinstein Co.           Original Screenplay             Comedy   
3197  Paramount Pictures     Based on Book/Short Story  Thriller/Suspense   
3198       Sony Pictures  Based on Comic/Graphic Novel          Adventure   
3199       Sony Pictures                        Remake          Adventure   
3200       Sony Pictures                        Remake          Adventure   

             Creative Type         Director  Rotten Tomatoes Rating  \
0                      NaN              NaN                     NaN   
1                      NaN              NaN                     NaN   
2                      NaN              NaN                     NaN   
3                      NaN              NaN                    13.0   
4     Contemporary Fiction              NaN                    62.0   
...                    ...              ...                     ...   
3196  Contemporary Fiction      Kevin Smith                    65.0   
3197         Dramatization    David Fincher                    89.0   
3198            Super Hero     Peter Hewitt                     3.0   
3199    Historical Fiction  Martin Campbell                    26.0   
3200    Historical Fiction  Martin Campbell                    82.0   

      IMDB Rating  IMDB Votes  
0             6.1      1071.0  
1             6.9       207.0  
2             6.8       865.0  
3             NaN         NaN  
4             3.4       165.0  
...           ...         ...  
3196          7.0     55687.0  
3197          NaN         NaN  
3198          3.4      7424.0  
3199          5.7     21161.0  
3200          6.7      4789.0  

[3201 rows x 16 columns]

### US Gross

In [80]:
# ADICIONE O SEU CÓDIGO AQUI
MeanGross = filmes["US Gross"].mean()
MedianGross = filmes["US Gross"].median()
DesvioGross = filmes["US Gross"].std()

print("Media US Gross: %f" %(MeanGross))
print("Mediana US Gross: %f" %(MedianGross))
print("Desvio Padrão US Gross: %f" %(DesvioGross))

Media US Gross: 44002085.163745
Mediana US Gross: 22019465.500000
Desvio Padrão US Gross: 62555311.390662


### Production Budget

In [68]:
# ADICIONE O SEU CÓDIGO AQUI
MeanBudget = filmes["Production Budget"].mean()
MedianBudget = filmes["Production Budget"].median()
DesvioBudget = filmes["Production Budget"].std()

print("Media Production Budget: %f" %(MeanBudget))
print("Mediana Production Budget: %f" %(MedianBudget))
print("Desvio Padrão Production Budget: %f" %(DesvioBudget))

Media Production Budget: 31069171.448438
Mediana Production Budget: 20000000.000000
Desvio Padrão Production Budget: 35585913.444644


### Running Time

In [78]:
# ADICIONE O SEU CÓDIGO AQUI
MeanTime = filmes["Running Time (min)"].mean()
MedianTime = filmes["Running Time (min)"].median()
DesvioTime = filmes["Running Time (min)"].std()

print("Media Running Time: %f" %(MeanTime))
print("Mediana Running Time: %f" %(MedianTime))
print("Desvio Padrão Running Time: %f" %(DesvioTime))

Media Running Time: 110.193548
Mediana Running Time: 107.000000
Desvio Padrão Running Time: 20.171014


### IMDB Rating

In [74]:
# ADICIONE O SEU CÓDIGO AQUI
MeanRating = filmes["IMDB Rating"].mean()
MedianRating = filmes["IMDB Rating"].median()
DesvioRating = filmes["IMDB Rating"].std()

print("Media IMDB Rating: %f" %(MeanRating))
print("Mediana IMDB Rating: %f" %(MedianRating))
print("Desvio Padrão IMDB Rating: %f" %(DesvioRating))

Media IMDB Rating: 6.283467
Mediana IMDB Rating: 6.400000
Desvio Padrão IMDB Rating: 1.252290


### IMDB Votes

In [76]:
# ADICIONE O SEU CÓDIGO AQUI
MeanVotes = filmes["IMDB Votes"].mean()
MedianVotes = filmes["IMDB Votes"].median()
DesvioVotes = filmes["IMDB Votes"].std()

print("Media IMDB Votes: %f" %(MeanVotes))
print("Mediana IMDB Votes: %f" %(MedianVotes))
print("Desvio Padrão IMDB Votes: %f" %(DesvioVotes))

Media IMDB Votes: 29908.644578
Mediana IMDB Votes: 15106.000000
Desvio Padrão IMDB Votes: 44937.582335


### Análise dos Resultados

Quais colunas tiveram valores muito diferentes entre média e mediana? Quais os seus desvios padrão?

**RESPONDA AQUI**

In [ ]:
R -> IMDB Votes, Production Budget e US Gross.

Que conclusões podemos tirar desses resultados?

**RESPONDA AQUI**

R->   A partir das análises das médias e medianas é perceptível que na maioria dos casos a media e a mediana possuem valores 
"próximos". Além disso,a análise exploratória proporcionou uma visão geral de que por exmeplo, a maioria dos filmes 
se pagam, pois a média do valor da bilheteria (Us Gross) é maior que a média do valor do orçamento (Production Budget).
      Com base neste DataFrame é possível afirmar também que além de um filme se pagar, ele também vai ser "bem aceito" pela crítica tendo em vista que a média das notas do IMDB é (Media IMDB Rating: 6.283467). Porém, o desvio padrão para o IMDB Rating é baixo (quanto mais próximo de zero, melhor). Ou seja, quanto mais próximo de zero o desvio for, significa que os dados estão mais homogêneos.